test/decorators/ConfigClass.spec.ts

```diff
@@ -724,7 +724,93 @@ describe('ConfigClass', () => {
       chai.expect((<any>c2.subArr[0]).toJSON()).to.deep.equal({num: 100});
     });
 
+    it('should load dynamically added class from partial config', async () => {
+  
+      @SubConfigClass()
+      class Inner {
+        @ConfigProperty()
+        b: number = 3;
+      }
+  
+  
+      @SubConfigClass()
+      class Sub {
+        @ConfigProperty()
+        subNum: number = 3;
+  
+        @ConfigProperty()
+        inner: unknown = {};
```

Comment:
This is not the best approach.

I'm strugling actually with dynamically changing the config setup. (what we actually need for extensions).

The best so far is that I created  a generic config type that gives better support to dynamic changes: https://github.com/bpatrik/typeconfig/blob/master/src/GenericConfigType.ts

Reply:
Here you can find some tests with it: test/decorators/ConfigPropoerty.spec.ts